### PR TITLE
Convert registration methods simplified and additional convert helper implementations added.

### LINKED ISSUE #91: Simplify ConvertRegistration methods
create to many / to one methods have generic parameters TTargetValue and TConcreteTargetValue.
Reason: TConcreteTargetValue is used to create a concrete object - a parameterless constructor is assumed.
But how a concrete target value is instantiated should be the responsibility of the CreateConvertHelper class.
The type parameters of convert registration methods should be reduced to TTargetValue with class constraint.

When upgrading to the version including this fix you have to:

- include BBT.StructueTools.Extensions into your projects
- register the derived IConvertHelperFactory and ICreateConvertHelper interfaces and generic implementations in your IoC container
- change IConvertHelperFactory respectively ICreateConvertHelper dependencies to the more specific interfaces
- if necessary reduce the generic type parameters of ICreateConverHelper and factory (TConcreteTarget is partially removed)

### LINKED ISSUE #93: Introduce ICreateConvertFromStrategyHelper and ICreateTargetImplConvertTargetImplHelper
Add two additiona ICreateConvertHelper implementations for the following use cases:
- delegate the create and convert to a strategy
- register and resolve the convert registrations based on TTargetImpl instead of TTarget

### PR DESCRIPTION
**Pull request by stefan-lindegger**
Fixes #91 
Fixes #93
- Convert registration methods simplified. Generic parameter TConcreteTarget removed.
- CreateConvertHelper and -Factory implementations moved to BBT.StructureTools.Extensions project.
- IoC registrations removed (must be done in calling code).
- Additional CreateConvertHelper and Factory implementations added
  - ICreateTargetImplConvertTargetImplHelper
  - ICreateConvertFromStrategyHelper

When upgrading to this version you have to:
- include BBT.StructueTools.Extensions into your projects
- register the derived IConvertHelperFactory and ICreateConvertHelper interfaces and generic implementation in your IoC container
- change IConvertHelperFactory respectively ICreateConvertHelper dependencies to the more specific interfaces
- if necessary reduce the generic type parameters of ICreateConverHelper and factory (TConcreteTarget is partially removed)

### CODE REVIEW

File: src/BBT.StructureTools.Extensions/BBT.StructureTools.Extensions.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <PackageLicenseExpression>MIT</PackageLicenseExpression>
+    <Copyright>Copyright © BBT Software AG</Copyright>
+    <Description>Extensions to BBT.StructureTools (e.g. implementations of ICreateConvertHelper and IConvertHelperFactory interfaces)</Description>
+    <PackageIcon></PackageIcon>
+    <RepositoryUrl>https://github.com/bbtsoftware/BBT.StructureTools.git</RepositoryUrl>
+    <RepositoryType>git</RepositoryType>
+    <PackageTags>copy compare convert</PackageTags>
+    <PackageReleaseNotes>https://github.com/bbtsoftware/BBT.StructureTools/releases/tag/1.0.4</PackageReleaseNotes>
+  </PropertyGroup>
```
Comment (marco-bertschi) [inline src/BBT.StructureTools.Extensions/BBT.StructureTools.Extensions.csproj:26]: Have these properties the same as [here](https://github.com/bbtsoftware/BBT.StructureTools/blob/develop/src/BBT.StructureTools/BBT.StructureTools.csproj)

File: src/BBT.StructureTools.sln [hunk 2]
```diff
@@ -1,12 +1,14 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.28307.271
+# Visual Studio Version 16
```
Comment (marco-bertschi) [inline src/BBT.StructureTools.sln]: Is that done on porpoise?

**Review by marco-bertschi — CHANGES_REQUESTED** [review]
Please adapt at least the property group in the new project

File: src/BBT.StructureTools.Extensions/BBT.StructureTools.Extensions.csproj (hunk 1 above)
Comment (stefan-lindegger) [reply src/BBT.StructureTools.Extensions/BBT.StructureTools.Extensions.csproj:26]: Done.

File: src/BBT.StructureTools.sln (hunk 2 above)
Comment (stefan-lindegger) [reply src/BBT.StructureTools.sln]: No. Changed to the original version.

**Review by marco-bertschi — APPROVED** [review]
LGTM

**Pull request merged**